packages/fungible-entangler/src/index.ts

```diff
@@ -0,0 +1,856 @@
+      "FungibleChildEntanglerV0",
+      account.data
+    );
+
+    return {
+      ...coded,
+      publicKey: pubkey,
+    };
+  };
+
+  getChildEntangler(
+    entanglerKey: PublicKey
+  ): Promise<IFungibleChildEntangler | null> {
+    return this.getAccount(entanglerKey, this.childEntanglerDecoder);
+  }
+
+  async createFungibleParentEntanglerInstructions({
+    authority = this.provider.wallet.publicKey,
+    payer = this.provider.wallet.publicKey,
+    source = this.provider.wallet.publicKey,
+    mint,
+    dynamicSeed,
+    amount,
+    goLiveDate = new Date(new Date().valueOf() - 10000), // 10 secs ago
+    freezeSwapDate,
+  }: ICreateFungibleParentEntanglerArgs): Promise<
+    InstructionResult<ICreateFungibleParentEntanglerOutput>
+  > {
+    const mintAcct = await getMintInfo(this.provider, mint);
+    const sourceAcct = await this.provider.connection.getAccountInfo(source);
+    amount = toNumber(amount, mintAcct);
+
+    // Source is a wallet, need to get the ATA
+    if (!sourceAcct || sourceAcct.owner.equals(SystemProgram.programId)) {
+      const ataSource = await Token.getAssociatedTokenAddress(
+        ASSOCIATED_TOKEN_PROGRAM_ID,
+        TOKEN_PROGRAM_ID,
+        mint,
+        payer,
+        true
+      );
+
+      if (!(await this.accountExists(ataSource))) {
+        throw new Error(
+          `Owner of ${payer?.toBase58()} does not hold any ${mint.toBase58()} tokens`
+        );
+      }
+
+      source = ataSource;
+    }
+
+    const sourceAcctAta = await getTokenAccount(this.provider, source);
+    const instructions: TransactionInstruction[] = [];
+    const signers: Keypair[] = [];
+
+    const [entangler, _entanglerBump] =
+      await FungibleEntangler.fungibleParentEntanglerKey(mint, dynamicSeed);
+
+    const [storage, _storageBump] = await FungibleEntangler.storageKey(
+      entangler
+    );
+
+    instructions.push(
+      await this.instruction.initializeFungibleParentEntanglerV0(
+        {
+          authority,
+          dynamicSeed,
+          goLiveUnixTime: new BN(Math.floor(goLiveDate.valueOf() / 1000)),
+          freezeSwapUnixTime: freezeSwapDate
+            ? new BN(Math.floor(freezeSwapDate.valueOf() / 1000))
+            : null,
+        },
+        {
+          accounts: {
+            payer,
+            entangler,
+            parentStorage: storage,
+            parentMint: mint,
+            tokenProgram: TOKEN_PROGRAM_ID,
+            systemProgram: SystemProgram.programId,
+            rent: SYSVAR_RENT_PUBKEY,
+            clock: SYSVAR_CLOCK_PUBKEY,
+          },
+        }
+      ),
+      Token.createTransferInstruction(
+        TOKEN_PROGRAM_ID,
+        source,
+        storage,
+        sourceAcctAta.owner,
+        [],
+        new u64(
+          (amount * Math.pow(10, mintAcct.decimals)).toLocaleString(
+            "fullwide",
+            {
+              useGrouping: false,
+            }
+          )
+        )
+      )
```

Comment:
This is the one part i'm unsure of. Should this transfer happen at the program level?